### PR TITLE
added devicePath check for fcDisk

### PR DESCRIPTION
**Pull request by NikhilSharmaWe**
#### What type of PR is this?
Bug Fixer 

#### What this PR does / why we need it:
It adds device path check for fc disk
#### Which issue(s) this PR fixes:

part of #107231 

#### Does this PR introduce a user-facing change?

```release-note
none
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/volume/fc/fc_util.go [hunk 1]
```diff
@@ -254,7 +254,12 @@ func (util *fcUtil) AttachDisk(b fcDiskMounter) (string, error) {
 		return "", err
 	}
 
-	return devicePath, nil
+	exists, err := mount.PathExists(devicePath)
+	if exists && err == nil {
+		return devicePath, nil
+	} else {
+		return "", err
```
Comment (jsafrane) [inline pkg/volume/fc/fc_util.go:261]: It should return an actual error instead of nil when `exists` is false and `err` is nil.

**Review by jsafrane — COMMENTED** [review]
I don't think it fixes any actual bug, but it should help with general robustness of iSCSI code.

File: pkg/volume/fc/fc_util.go (hunk 1 above)
Comment (NikhilSharmaWe) [reply pkg/volume/fc/fc_util.go:261]: @jsafrane do you want to say when `exists` is true and `err` is nil, then we should return err ?

Comment (jsafrane) [reply pkg/volume/fc/fc_util.go:261]: No, when `exists` is false (= returned device does not exist), then return error even when `err` is nil.

Comment (NikhilSharmaWe) [reply pkg/volume/fc/fc_util.go:261]: @jsafrane Sorry for asking again, Do I have to make another condition ?
```
if exists == false && err == nil{
     return "", nil
}
```

Comment (NikhilSharmaWe) [reply pkg/volume/fc/fc_util.go:261]: @jsafrane Ok, so I have to add 
```
if exists == false && err == nil{
    return "", err
}
```
If it is correct, why we are returning err at place of nil ? And why are we not doing it with `exists && err == nil`.

Comment (jsafrane) [reply pkg/volume/fc/fc_util.go:264]: No, if `exists == false`, you should return a *new error*, like `fmt.Errorf("device %s does not exist", devicePath)`. Why else do you add a check for `devicePath`, if you don't use the result?

**Pull request merged**